File: modules/wear-bridge/android/src/main/java/expo/modules/wearbridge/WearBridgeModuleSharedObject.kt
package expo.modules.wearbridge

import expo.modules.kotlin.AppContext
import expo.modules.kotlin.sharedobjects.SharedObject

class WearBridgeModuleSharedObject(appContext: AppContext) : SharedObject(appContext) {
  var count: Int = 0

  override fun sharedObjectDidRelease() {
    super.sharedObjectDidRelease()
  }
}

File: modules/wear-bridge/src/WearBridgeModuleSharedObject.ts
import { SharedObject, useReleasingSharedObject } from 'expo-modules-core';

import WearBridgeModule from './WearBridgeModule';

export declare class WearBridgeModuleSharedObject extends SharedObject {
  count: number;
}

/**
 * Creates a new WearBridgeModuleSharedObject instance.
 * You are responsible for releasing it from memory by calling `release()` when done.
 */
export function createWearBridgeModuleSharedObject(): WearBridgeModuleSharedObject {
  return new WearBridgeModule.WearBridgeModuleSharedObject();
}

/**
 * A hook that creates a WearBridgeModuleSharedObject instance and automatically
 * releases it when the component unmounts.
 */
export function useWearBridgeModuleSharedObject(): WearBridgeModuleSharedObject {
  return useReleasingSharedObject(() => new WearBridgeModule.WearBridgeModuleSharedObject(), []);
}
